src/cascadia/TerminalSettingsModel/WslDistroGenerator.cpp

```diff
@@ -136,3 +143,201 @@ std::vector<Profile> WslDistroGenerator::GenerateProfiles()
+// Arguments:
+// - wslRootKey: the HKEY for the Lxss node.
+// - guid: the string representation of the GUID for the distro to inspect
+// Return Value:
+// - the HKEY if it exists and we can read it, else nullptr
+static HKEY openDistroKey(const wil::unique_hkey& wslRootKey, const std::wstring& guid)
+{
+    HKEY hKey{ nullptr };
+    if (RegOpenKeyEx(wslRootKey.get(), guid.c_str(), 0, KEY_ALL_ACCESS, &hKey) == ERROR_SUCCESS)
+    {
+        return hKey;
+    }
+    return nullptr;
+}
+
+// Function Description:
+// - Get the list of all the guids of all the WSL distros from the registry. If
+//   we fail to open or read the root reg key, we'll return false.
+//   Places the guids of all the distros into the "guidStrings" param.
+// Arguments:
+// - wslRootKey: the HKEY for the Lxss node.
+// - names: a vector that receives all the guids of the installed distros.
+// Return Value:
+// - false if we failed to enumerate all the WSL distros
+static bool getWslGuids(const wil::unique_hkey& wslRootKey,
+                        std::vector<std::wstring>& guidStrings)
+{
+    if (!wslRootKey)
+    {
+        return false;
+    }
+
+    // Figure out how many subkeys we have, and what the longest name of these subkeys is.
+    DWORD dwNumSubKeys = 0;
+    if (RegQueryInfoKey(wslRootKey.get(),
+                        nullptr,
+                        nullptr,
+                        nullptr,
+                        &dwNumSubKeys,
+                        nullptr,
+                        nullptr,
+                        nullptr,
+                        nullptr,
+                        nullptr,
+                        nullptr,
+                        nullptr) != ERROR_SUCCESS)
+    {
+        return false;
+    }
+
+    wchar_t buffer[39]; // a {GUID} is 38 chars long
+    for (DWORD i = 0; i < dwNumSubKeys; i++)
+    {
+        DWORD length;
+        const auto result = RegEnumKeyEx(wslRootKey.get(), i, &buffer[0], &length, nullptr, nullptr, nullptr, nullptr);
+        if (result == ERROR_NO_MORE_ITEMS)
+        {
+            break;
+        }
+
+        if (result == ERROR_SUCCESS &&
+            length == 38 &&
+            buffer[0] == L'{' &&
+            buffer[37] == L'}')
+        {
+            guidStrings.emplace_back(&buffer[0], length);
+        }
+    }
+
+    return true;
+}
+
+// Function Description:
+// - Get the list of all the names of all the WSL distros from the registry. If
+//   we fail to open any regkey for the GUID of a distro, we'll just skip it.
+//   Places the names of all the distros into the "names" param.
+// Arguments:
+// - wslRootKey: the HKEY for the Lxss node.
+// - guidStrings: A list of all the GUIDs of the installed distros
+// - names: a vector that receives all the names of the installed distros.
+// Return Value:
+// - false if the root key was invalid, else true.
+static bool getWslNames(const wil::unique_hkey& wslRootKey,
+                        std::vector<std::wstring>& guidStrings,

```

Comment:
We consume this one `const`ly, right?

```suggestion
                        const std::vector<std::wstring>& guidStrings,
```